src/mgr/Inquirer.cc

```diff
@@ -73,17 +74,22 @@ Mgr::Inquirer::start()
     Must(Comm::IsConnOpen(conn));
     Must(aggrAction != nullptr);
 
+    const auto &origin = aggrAction->command().params.httpOrigin;
+    const auto originOrNil = origin.size() ? origin.termedBuf() : nullptr;
+
     std::unique_ptr<MemBuf> replyBuf;
     if (strands.empty()) {
         const char *url = aggrAction->command().params.httpUri.termedBuf();
         const auto mx = MasterXaction::MakePortless<XactionInitiator::initIpc>();
         auto *req = HttpRequest::FromUrlXXX(url, mx);
         ErrorState err(ERR_INVALID_URL, Http::scNotFound, req, nullptr);
         std::unique_ptr<HttpReply> reply(err.BuildHttpReply());
+        CacheManager::PutCommonResponseHeaders(*reply, originOrNil);
         replyBuf.reset(reply->pack());
     } else {
         std::unique_ptr<HttpReply> reply(new HttpReply);
         reply->setHeaders(Http::scOkay, nullptr, "text/plain", -1, squid_curtime, squid_curtime);
+        CacheManager::PutCommonResponseHeaders(*reply, originOrNil);
```

Comment:
Note that these Inquerer.cc changes added missing Access-Control-* response headers to cache manager responses in SMP mode.